changelog.d/6509.bugfix

```diff
@@ -0,0 +1 @@
+Fix pusher worker failing because it can't retrieve retention policies for rooms.
```

Comment:
Oh, ideally this would say "Fix regression where sending email push did not work when using a pusher worker" or somethibg

Reply:
Can't we fix that when generating the next changelog?

Reply:
It'd be nice to fix it up now, as otherwise it puts quite a large burden on those doing a release

Reply:
Done in https://github.com/matrix-org/synapse/commit/ae49d29ef1dabec25dc348bf11c2919d28d1cef3